pkgs/development/compilers/kotlin/default.nix

```diff
@@ -1,43 +1,100 @@
 { lib, stdenv, fetchurl, makeWrapper, jre, unzip }:
 
-stdenv.mkDerivation rec {
-  pname = "kotlin";
+let
   version = "1.5.30";
+  homepage = "https://kotlinlang.org/";
+  license = lib.licenses.asl20;
+  description = "A modern programming language that makes developers happier";
+  longDescription = ''
+    Kotlin is a modern but already mature programming language aimed to make
+    developers happier. It’s concise, safe, interoperable with Java and other
+    languages, and provides many ways to reuse code between multiple platforms
+    for productive programming.
+  '';
+in {
+  default = stdenv.mkDerivation rec {
+    pname = "kotlin";
+    inherit version;
+
+    src = fetchurl {
+      url =
+        "https://github.com/JetBrains/kotlin/releases/download/v${version}/kotlin-compiler-${version}.zip";
+      sha256 = "sha256-Je69ubsuFl5LqO+/j/lDxF1Pw52//CwcqgWejdgTZ18=";
+    };
+
+    propagatedBuildInputs = [ jre ];
+    nativeBuildInputs = [ makeWrapper unzip ];
+
+    installPhase = ''
+      mkdir -p $out
+      rm "bin/"*.bat
+      mv * $out
 
-  src = fetchurl {
-    url = "https://github.com/JetBrains/kotlin/releases/download/v${version}/kotlin-compiler-${version}.zip";
-    sha256 = "sha256-Je69ubsuFl5LqO+/j/lDxF1Pw52//CwcqgWejdgTZ18=";
+      for p in $(ls $out/bin/) ; do
+        wrapProgram $out/bin/$p --prefix PATH ":" ${jre}/bin ;
+      done
+
+      if [ -f $out/LICENSE ]; then
+        install -D $out/LICENSE $out/share/kotlin/LICENSE
+        rm $out/LICENSE
+      fi
+    '';
+    meta = {
+      inherit homepage license description longDescription;
+      maintainers = with lib.maintainers; [ SubhrajyotiSen ];
+      platforms = lib.platforms.all;
+    };
   };
 
-  propagatedBuildInputs = [ jre ] ;
-  nativeBuildInputs = [ makeWrapper unzip ];
+  native = stdenv.mkDerivation rec {
+    pname = "kotlin-native";
+    inherit version;
 
-  installPhase = ''
-    mkdir -p $out
-    rm "bin/"*.bat
-    mv * $out
+    src = let
+      platform = if stdenv.isLinux then
+        "linux"
+      else if stdenv.isDarwin then
```

Comment:
Won't it be fine if we assume that if the system isn't Linux, it is Darwin?
We can skip the empty `else` block that way. Similarly for other checks as well

Reply:
I would expect someone to help with the windows packaging. UnU

It seems different so I didn't want to add it without testing.

Reply:
Not really, the only thing I can do is add the hashes and let it run as is.

Reply:
> Won't it be fine if we assume that if the system isn't Linux, it is Darwin?

No, we can't assume that.

Reply:
Is there a switch like statement on platform type?

Reply:
more like https://github.com/NixOS/nixpkgs/blob/956b9023e1e0d5d20eceda0e9493311ef57361d0/pkgs/applications/networking/instant-messengers/vk-messenger/default.nix#L22